### DISCUSSION TITLE
Consider enable of parallelism inside framework.

### DISCUSSION THREAD

**Discussion by lwasylow** [utPLSQL/utPLSQL#1332] [Feature Requests]
Consider enable of parallelism inside framework using dbms parallel, job_chains etc.
What are pros and cons.

**Comment by pesse** [comment]
Cons from my POV:
- Adds a lot of complexity test-wise, especially when we want to support it on numerous database versions
- Will probably make debugging and maintainability harder
- There is an easy workaround by running utPLSQL-cli multiple times on different suitepaths/tags

**Comment by jgebal** [comment]
> I had a look at junit5 about parallel and way its implemented there is the annotation declaration for suite or test methods. In their implementation it allows for permutations of parallel suites or test methods. 
> 
> I think we should start with an annotation on the suite to parallelize all tests under it. Author is responsible for making them paralell safe.
> 
> So the suites will call serially but tests within suites that is annotated would run parallel.
> 
> The results would have to be stored in the table as the chunks will be run by different sessions. 
> 
> The sample annotation couls looks like `--%parallel(4)`
> 
> 
> 
> In case of nested suite the parallel wil be inherited to child suites. The child suites will be run serial however the tests within these suites will run parallel as defined by parent suite.
> 
> 
> 
> We will have to make sure that coverage data will be linked to same runid or at least we are able to merge them from child session to coordinator session, same goes for reporting.  

 _Originally posted by @lwasylow in [#1144](https://github.com/utPLSQL/utPLSQL/issues/1144#issuecomment-1257072289)_
